mino/minows/mod.go

```diff
@@ -0,0 +1,127 @@
+package minows
+
+import (
+	"github.com/rs/zerolog"
+	"go.dedis.ch/dela"
+	"go.dedis.ch/dela/serde/json"
+	"regexp"
+	"strings"
+
+	"github.com/libp2p/go-libp2p"
+	"github.com/libp2p/go-libp2p/core/crypto"
+	"github.com/libp2p/go-libp2p/core/host"
+	"github.com/libp2p/go-libp2p/core/protocol"
+	ma "github.com/multiformats/go-multiaddr"
+	"go.dedis.ch/dela/mino"
+	"go.dedis.ch/dela/serde"
+	"golang.org/x/xerrors"
+)
+
+var pattern = regexp.MustCompile("^[a-zA-Z0-9]+$")
+
+// Minows
+// - implements mino.Mino
+type minows struct {
+	logger zerolog.Logger
+
+	myAddr   address
+	host     host.Host
+	segments []string
+	rpcs     map[string]any
+	factory  addressFactory
+}
+
+// NewMinows creates a new Minows instance that starts listening.
+// listen: listening address in multiaddress format,
+// e.g. /ip4/127.0.0.1/tcp/80/ws
+// public: public dial-able address in multiaddress format,
+// e.g. /dns4/p2p-1.c4dt.dela.org/tcp/443/wss
+// `public` can be nil and will be determined
+// by the listening address and the port the host has bound to.
+// key: private key representing this mino instance's identity
+func NewMinows(listen, public ma.Multiaddr, key crypto.PrivKey) (mino.Mino,
```

Comment:
Just based on this API, it would seem that `minows` really ended up being `minolibp2p`.

IIRC, we had discussed (a while back, with @ineiti) `minows` as being native websockets, and a first step towards having a full `minolibp2p`. Effectively, it seems the native websocket implementation doesn't exist, and only the libp2p one does. That's perfectly fine and seems highly configurable. 

I guess that, should we (later on) want to use a different setup / sets of protocols accessed through libp2p, the different setup could be created based on the same `minows` package, just configured differently.

Right?

Reply:
One thing that mildly worries me is that according to [this](https://multiformats.io/multiaddr/), the `Multiaddr` format is still a work-in-progress... 
which would make it a so-so candidate for integration in a top-level API.
Unless only the documentation is WIP? 🤔

Reply:
- `minows` vs. `minolibp2p` - we did check out Gorilla websockets (one of the many things Zhe didn't talk about), but in the end went with libp2p for the possibility of changing protocols later. So yes, we can change the protocol pretty easily. And another student project might create a generalized `minolibp2p` out of `minows`.
- `Multiaddr` WIP: As far as I can gather from your document, it is indeed the document which is WIP, but the format is more or less stable. They even talk about handing in an IETF RFC. So I think we're good here.